diff --git a/gym-STAR/gym_STAR/env/Fix.py b/gym-STAR/gym_STAR/env/Fix.py
index ee150ad..168c012 100644
--- a/gym-STAR/gym_STAR/env/Fix.py
+++ b/gym-STAR/gym_STAR/env/Fix.py
@@ -78,9 +78,9 @@ class Fix(gym.Env):
         self.STAR_position = [0, 0, 10]
         self.link_position = [0, 0, 0]
         self.type = np.zeros(shape=(self.K, 1))
-        self.P_K_list = np.random.normal(scale=100, size=(3, self.K))
-        self.P_K_list[:, :3] += 200
-        self.P_K_list[:, 3:] -= 200
+        self.P_K_list = np.random.uniform(low=-500, high=500, size=(3, self.K))
+        # self.P_K_list[:, :3] += 200
+        # self.P_K_list[:, 3:] -= 200
         self.P_K_list[2, :] = 0
         self.t = 0
 
@@ -268,9 +268,8 @@ class Fix(gym.Env):
     #TODO reset the environmrnt, user position, time, observation state, STAR position???
     def reset(self, seed=None, options=None):
         super().reset(seed=seed)
-        self.P_K_list = np.random.normal(scale=100, size=(3, self.K))
-        self.P_K_list[:, :3] += 200
-        self.P_K_list[:, 3:] -= 200
+        self.P_K_list = np.random.uniform(low=-500, high=500, size=(3, self.K))
+        # self.P_K_list[:, :] -= 500
         self.P_K_list[2, :] = 0
 
         # self.FD_B_K = np.random.normal(scale=1, size=(self.M, self.K, self.T)) + np.random.normal(scale=1, size=(self.M, self.K, self.T)) * 1j
diff --git a/gym-STAR/gym_STAR/env/Fix_Pos.py b/gym-STAR/gym_STAR/env/Fix_Pos.py
index df8fdf7..ccf665d 100644
--- a/gym-STAR/gym_STAR/env/Fix_Pos.py
+++ b/gym-STAR/gym_STAR/env/Fix_Pos.py
@@ -78,9 +78,8 @@ class Fix_Pos(gym.Env):
         self.STAR_position = [0, 0, 10]
         self.link_position = [0, 0, 0]
         self.type = np.zeros(shape=(self.K, 1))
-        self.P_K_list = np.random.normal(scale=100, size=(3, self.K))
-        self.P_K_list[:, :3] += 200
-        self.P_K_list[:, 3:] -= 200
+        self.P_K_list = np.random.uniform(low=-500, high=500, size=(3, self.K))
+        # self.P_K_list[:, :] -= 500
         self.P_K_list[2, :] = 0
         self.t = 0
 
@@ -272,9 +271,8 @@ class Fix_Pos(gym.Env):
     #TODO reset the environmrnt, user position, time, observation state, STAR position???
     def reset(self, seed=None, options=None):
         super().reset(seed=seed)
-        self.P_K_list = np.random.normal(scale=100, size=(3, self.K))
-        self.P_K_list[:, :3] += 200
-        self.P_K_list[:, 3:] -= 200
+        self.P_K_list = np.random.uniform(low=-500, high=500, size=(3, self.K))
+        # self.P_K_list[:, :] -= 500
         self.P_K_list[2, :] = 0
 
         # self.FD_B_K = np.random.normal(scale=1, size=(self.M, self.K, self.T)) + np.random.normal(scale=1, size=(self.M, self.K, self.T)) * 1j
diff --git a/ppo_normalizeLayer.py b/ppo_normalizeLayer.py
index 7a41adc..ec1d633 100644
--- a/ppo_normalizeLayer.py
+++ b/ppo_normalizeLayer.py
@@ -87,7 +87,7 @@ def make_env(env_id, idx, capture_video, run_name, gamma):
             env = gym.make(env_id, render_mode="rgb_array")
         else:
             env = gym.make(env_id)
-        env = gym.wrappers.FlattenObservation(env)  # deal with dm_control's Dict observation space
+        env = gym.wrappers.FlattenObservation(env)  # deal with dm`_control's Dict observation space
         env = gym.wrappers.RecordEpisodeStatistics(env)
         if capture_video:
             if idx == 0:
diff --git a/wandb/debug-cli.JinLab.log b/wandb/debug-cli.JinLab.log
index 2d9c106..561e2c2 100644
--- a/wandb/debug-cli.JinLab.log
+++ b/wandb/debug-cli.JinLab.log
@@ -554,3 +554,12 @@ requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='api.wandb.ai', port=4
 2023-04-25 18:44:09 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
 2023-04-25 18:44:09 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
 2023-04-25 18:44:09 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-27 16:37:07 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-27 16:37:07 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-27 16:37:07 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-28 22:05:09 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-28 22:05:09 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-28 22:05:09 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-29 20:44:29 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-29 20:44:29 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
+2023-04-29 20:44:29 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))': /api/5288891/envelope/
